app/commands/user_track/viewed_community_solution/create.rb

```diff
@@ -4,6 +4,8 @@ class UserTrack::ViewedCommunitySolution::Create
   initialize_with :user, :track, :solution
 
   def call
+    return if solution.user == user
```

Comment:
(Avoid an extra db lookup)

```suggestion
    return if solution.user_id == user.id
```